lib/repository.php

```diff
@@ -55,7 +55,12 @@ public static function add($entity, $save = true) {
 
     public static function edit($id, $entity) {
         if (array_key_exists($id, self::$entities)) {
-            self::$entities[$id] = array_merge(self::$entities[$id], $entity);
+            if (is_array(self::$entities[$id])) {
+                $data =  array_merge(self::$entities[$id], $entity);;
+            } else {
+                $data = $entity;
```

Comment:
When that's an object, don't we still need a merge? Otherwise, it is an overwrite.

Reply:
when it is an object it already comes with the necessary modifications unlike an array where only certain properties can be modified

Reply:
Ok but this makes the whole method quite confusing - imagine someone trying to use it - how dothey know if this method actually replaces the entity or updates the passed entity params only? I still suggest you unify - either do a full replace and require calling code to pass full entities or merge. Having a different business logic based on type of incoming parameters is a bad programming practice.

Reply:
The same thing has been done here and merged

https://github.com/cypht-org/cypht/pull/1132/files#diff-e8ea654d76e94bfce8824e84e96fec6cf4d3497b6e3d446ab8e1f81d8eb983ceR58